### PR TITLE
Implement Verifiers

### PR DESCRIPTION
**Pull request by jmcardon**
PR checklist:

* [x] Test coverage for the proposed changes
* [x] PR description contains example output from repl interaction or a snippet from unit test output
* [x] Any changes that could be relevant to users [have been recorded in the changelog](https://github.com/kadena-io/pact-core/blob/master/CHANGELOG.md)

Additionally, please justify why you should or should not do the following:

* [x] Confirm replay/back compat
* [x] Benchmark regressions
* [x] (For Kadena engineers) Run integration-tests against a Chainweb built with this version of Pact

### CODE REVIEW

File: pact-tests/pact-tests/caps.repl [hunk 1]
```diff
@@ -955,3 +955,5 @@
 
 (env-sigs [{"key":"jose", "caps":[(C1 1) (C2 1) (C3 1)]}])
 (expect "managed capabilities compose correctly" 111 (run))
+(commit-tx)
```
Comment (rsoeldner) [inline pact-tests/pact-tests/caps.repl:958]: Just to confirm, this was _just_ missing but is not influence the outcome of the tests

File: pact-tests/pact-tests/coin-v5.repl [hunk 2]
```diff
@@ -931,6 +931,11 @@
   (map (remove 'module-hash) (env-events true)))
 
 (test-capability (coin.GAS))
+; note: DEBIT gets injected by chainweb.
```
Comment (rsoeldner) [inline pact-tests/pact-tests/coin-v5.repl:934]: 👍

File: pact/Pact/Core/IR/Eval/CEK.hs [hunk 3]
```diff
@@ -1479,9 +1494,11 @@ applyContToValue (BuiltinC env info frame cont) handler cv = do
             let acc' = PObject . _objectData <$> reverse acc
             in returnCEKValue cont handler (VList (V.fromList acc'))
     _ -> returnCEK cont handler (VError "higher order apply did not return a pactvalue" info)
-applyContToValue (CapBodyC cappop env info mcap mevent capbody cont) handler _ = do
+-- applyContToValue (CapBodyC cappop env info mcap mevent capbody cont) handler _ = do
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/CEK.hs:1497]: cleanup

File: pact/Pact/Core/IR/Eval/Runtime/Types.hs [hunk 4]
```diff
@@ -90,6 +87,10 @@ module Pact.Core.IR.Eval.Runtime.Types
  , CoreCEKValue
  , CoreEvalResult
  , EvalCapType(..)
+ , CapBodyState(..)
+ -- Todo: turn into individual lenses
+ -- this is purely a lazy export
```
Comment (rsoeldner) [inline pact/Pact/Core/IR/Eval/Runtime/Types.hs:92]: todo

File: pact/Pact/Core/Repl/Runtime/ReplBuiltin.hs [hunk 5]
```diff
@@ -286,6 +287,27 @@ envSigs info b cont handler _env = \case
...
+  [VList ks] ->
+    case traverse verifCapObj ks of
+      Just sigs -> do
+        (replEvalEnv . eeMsgVerifiers) .= M.fromList (V.toList sigs)
+        returnCEKValue cont handler $ VString "Setting transaction verifiers/caps"
+      Nothing -> returnCEK cont handler (VError ("env-verifiers: Expected object with 'name': string, 'caps': [capability]") info)
+    where
+    verifCapObj = \case
+      PObject o -> do
+        keyRaw<- M.lookup (Field "name") o
```
Comment (rsoeldner) [inline pact/Pact/Core/Repl/Runtime/ReplBuiltin.hs:301]:
```suggestion
        keyRaw <- M.lookup (Field "name") o
```

**Review by rsoeldner — APPROVED** [review]

File: pact-tests/pact-tests/caps.repl (hunk 1 above)
Comment (jmcardon) [reply pact-tests/pact-tests/caps.repl:958]: Indeed.

**Pull request merged**